### ISSUE TITLE
NaN during rnn training

### ISSUE THREAD

**Issue by moberc** [deeplearning4j/deeplearning4j#4488]
#### Issue Description

I am getting NaN frequently during rnn training.
Training cycles start OK, cost function is decreasing and then I am getting NaN half of the time.
I have tried changes to hyperparameters.
My counterpart is working with TensorFlow and does not have these issues.  Anything I could do?
Thx!

#### Version Information
0.9.1
Windows

**Comment by AlexDBlack** [comment]
I'm closing this - you have given no information that would allow us to even begin to identify/debug a cause to this, or provide specific advice.

In general, differences any of the hyperparameters (notably, learning rate, weight initialization, loss function, regularization, etc) as well as normalization can cause this.

I can suggest reading:
https://deeplearning4j.org/troubleshootingneuralnets
https://deeplearning4j.org/visualization

**Comment by moberc** [comment]
By the way, I was able to solve the issue by shuffling the examples in the training set.
